### PR TITLE
chore(rust): Bump to 1.72.1

### PR DESCRIPTION
**Pull request by jamilbk**
Fixes an issue pulling the `1.70-slim` image. Also introduces a few compiler fixes/enhancements.

Changelog:

https://releases.rs/docs/1.72.0/
https://releases.rs/docs/1.72.1/

### CODE REVIEW

File: rust/Cargo.toml [hunk 1]
```diff
@@ -13,6 +13,8 @@ members = [
   "connlib/gateway",
 ]
 
+resolver = "2"
```
Comment (jamilbk) [inline rust/Cargo.toml:16]: Fixes a warning about resolver ambiguity

**Review by thomaseizinger — APPROVED** [review]

**Review by conectado — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Formatting seems to be off.

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -36,7 +36,8 @@ jobs:
       - name: Start docker compose in the background
         run: docker compose up -d
       - name: Test that client can ping resource
-        run: docker compose exec -it client ping 172.20.0.100 -c 10
+        run: docker compose exec -it client timeout 60 bash -c 'until ping -W 1 -c 1 172.20.0.100 &>/dev/null; do true; done'
```
Comment (jamilbk) [inline .github/workflows/integration-tests.yml:39]: @conectado I went ahead and applied the `ping until success or timeout` logic here as we chatted about :-).

File: elixir/apps/domain/lib/domain/auth.ex [hunk 3]
```diff
@@ -611,12 +611,13 @@ defmodule Domain.Auth do
...
-         user_agent,
-         remote_ip
+         {:identity, identity_id, _context_payload},
+         _user_agent,
+         _remote_ip
        ) do
     with {:ok, identity} <- fetch_active_identity_by_id(identity_id),
-         true <- context_payload == session_context_payload(remote_ip, user_agent),
+         # XXX: Don't pin tokens to remote_ip and user_agent -- use device external_id instead?
+         # true <- context_payload == session_context_payload(remote_ip, user_agent),
```
Comment (jamilbk) [inline elixir/apps/domain/lib/domain/auth.ex]: Somehow this snuck back in. Leaving it commented out for now until we come up with some better params to pin the session token to.

**Comment by jamilbk** [comment]
Merged to #2195

**Pull request merged**